ppml/README.md

```diff
@@ -0,0 +1,505 @@
+      ```
+      To check the logs of an Spark executor, run
+      ```
+      sudo kubectl logs $( sudo kubectl get pod | grep "simplequery-.*-exec" -m 1 | cut -d " " -f1 )
+      ```
+  
+  4. If you setup [PPML Monitoring](docs/prepare_environment.md#optional-k8s-monitioring-setup), you can check PPML Dashboard to monitor the status in http://kubernetes_master_url:3000
+
+    ![image](https://user-images.githubusercontent.com/61072813/179948818-a2f6844f-0009-49d1-aeac-2e8c5a7ef677.png)
+
+  </details>
+<br />
+
+#### Step 4. Decrypt and Read Result
+When the job is done, you can decrypt and read result of the job. More details in [Decrypt Job Result](./services/kms-utils/docker/README.md#3-enroll-generate-key-encrypt-and-decrypt).
+
+  ```
+  docker exec -i $KMSUTIL_CONTAINER_NAME bash -c "bash /home/entrypoint.sh decrypt $appid $appkey $input_path"
+  ```
+
+### 3.3 More BigDL PPML Examples
+In addition to the above Spark Pi and Python HelloWorld programs running locally, and simplequery application running on the k8s cluster, we also provide other examples including Trusted Data Analysis, Trusted ML, Trusted DL and Trusted FL. You can find these examples in [more examples](./docs/examples.md). 
+
+## 4. Develop your own Big Data & AI applications with BigDL PPML
+
+We provide both Scala and Python APIs to help you develop your own applications with BigDL PPML. 
+
+First thing you need to do is create a `PPMLContext`, who wraps a `SparkSession` and provides read functions to read encrypted data files to plain-text RDD or DataFrame, also provides write functions to save DataFrame to encrypted data files. 
+
+If you are familiar with Spark, then you will find that using `PPMLConext` is very similar to Spark.
+
+### 4.1 PPMLContext In Scala
+
+#### 4.1.1 create a PPMLContext
+
+1. `create a PPMLContext with appName`
+
+   This is the simplest way to create a `PPMLContext`. When you don't need to read/write encrypted files, you can use this way to create a `PPMLContext`.
+
+   Code Example
+
+   ```scala
+   import com.intel.analytics.bigdl.ppml.PPMLContext
+   
+   val sc = PPMLContext.initPPMLContext("MyApp")
+   ```
+
+   If you want to read/write encrypted files, then you need to provide more information.
+
+2. `create a PPMLContext with appName & ppmlArgs`
+
+   `ppmlArgs` is a Map, what parameters in the Map depends on what kind of `KeyManagementService(KMS)` you use. we support 3 types of KMS ——`SimpleKeyManagementService`, `EHSMKeyManagementService`, `AzureKeyManagementService`. 
+
+   We use KMS to generate a `primaryKey` and `dataKey`, then we use the keys to encrypt/decrypt data.
+
+   > How to use KMS to generate primaryKey and dataKey, please refer to [this](https://github.com/intel-analytics/BigDL/blob/main/ppml/services/kms-utils/docker/README.md)
+
+   After preparing the keys, now you can create a `PPMLContext` with the following parameters.
```

Comment:
`ppmlArgs` is ppml arguments in a Map, `ppmlArgs` varies according to the kind of Key Management Service (KMS) you are using. Key Management Service (KMS) is used to generate `primaryKey` and `dataKey` to encrypt/decrypt data. We provide 3 types of KMS ——SimpleKeyManagementService, EHSMKeyManagementService, AzureKeyManagementService. 

   Refer to [KMS Utils](https://github.com/intel-analytics/BigDL/blob/main/ppml/services/kms-utils/docker/README.md) to use KMS to generate `primaryKey` and `dataKey`, then you are ready to create **PPMLContext** with `ppmlArgs`.